aiokafka/group_coordinator.py

```diff
@@ -0,0 +1,847 @@
+                                    Errors.NotCoordinatorForGroupError,
+                                    Errors.RequestTimedOutError):
+                    log.info(
+                        "OffsetCommit failed for group %s due to a"
+                        " coordinator error (%s), will find new coordinator"
+                        " and retry", self.group_id, error_type.__name__)
+                    self.coordinator_dead()
+                    raise error_type()
+                elif error_type in (Errors.UnknownMemberIdError,
+                                    Errors.IllegalGenerationError,
+                                    Errors.RebalanceInProgressError):
+                    # need to re-join group
+                    error = error_type(self.group_id)
+                    log.error(
+                        "OffsetCommit failed for group %s due to group"
+                        " error (%s), will rejoin", self.group_id, error)
+                    self._subscription.mark_for_reassignment()
+                    raise error
+                else:
+                    log.error(
+                        "OffsetCommit failed for group %s on partition %s"
+                        " with offset %s: %s", self.group_id, tp, offset,
+                        error_type.__name__)
+                    raise error_type()
+
+        if unauthorized_topics:
+            log.error("OffsetCommit failed for unauthorized topics %s",
+                      unauthorized_topics)
+            raise Errors.TopicAuthorizationFailedError(unauthorized_topics)
+
+    @asyncio.coroutine
+    def _maybe_auto_commit_offsets_sync(self):
+        if not self._enable_auto_commit:
+            return
+
+        yield from self.commit_offsets(
+            self._subscription.all_consumed_offsets()
+        )
+
+    @asyncio.coroutine
+    def auto_commit_routine(self, interval):
+        while not self._closing.done():
+            if (yield from self.coordinator_unknown()):
+                log.debug(
+                    "Cannot auto-commit offsets because the coordinator is"
+                    " unknown, will retry after backoff")
+                yield from asyncio.sleep(
+                    self._retry_backoff_ms / 1000.0, loop=self.loop)
+                continue
+
+            yield from asyncio.wait(
+                [self._closing], timeout=interval, loop=self.loop)
+
+            # select offsets that should be committed
+            offsets = {}
+            for partition in self._subscription.assigned_partitions():
+                tp = self._subscription.assignment[partition]
+                if tp.position != tp.committed and tp.has_valid_position:
+                    offsets[partition] = OffsetAndMetadata(tp.position, '')
+
+            try:
+                yield from self.commit_offsets(offsets)
+            except Errors.KafkaError as error:
+                if error.retriable:
```

Comment:
How about we change this to something like:
```Python
if error.retriable and not self._closing.done():
```
So if we close and failed to commit - we will see it in logs